### ISSUE TITLE
TERMinating a vw daemon generates a malloc error

### ISSUE THREAD

**Issue by joachimdb** [JohnLangford/vowpal_wabbit#626]
Joachims-MacBook-Air:resources joachim$ vw --daemon --pid_file /tmp/vw.pid
Num weight bits = 18
learning rate = 0.5
initial_t = 0
power_t = 0.5
using no cache
calling accept

Joachims-MacBook-Air:resources joachim$ cat /tmp/vw.pid | xargs kill

finished run
number of examples = 0
weighted example sum = 0
weighted label sum = 0
average loss = nan
total feature number = 0
vw(19420,0x7fff74ba6300) malloc: *** error for object 0x10389d000: pointer being freed was not allocated
*** set a breakpoint in malloc_error_break to debug

**Comment by joachimdb** [comment]
Update: this issue cannot be reproduced on Ubuntu

**Comment by JohnLangford** [comment]
On 05/20/2015 06:10 AM, Joachim De Beule wrote:
>
> Update: this issue cannot be reproduced on Ubuntu
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/JohnLangford/vowpal_wabbit/issues/626#issuecomment-103834248>.
>

I'm not sure what to do about this one.   The termination process in 
daemon mode is irregular.  If the parent process gets a sigterm, it 
enters this bit:
https://github.com/JohnLangford/vowpal_wabbit/blob/master/vowpalwabbit/parser.cc#L539
killing each child.  Maybe something goes wrong?  But I don't have a 
mac, so I can't reproduce personally.  Someone else will need to 
investigate.

-John

**Comment by JohnLangford** [comment]
Closing for now---I can't replicate personally.  I'd certainly welcome a pull request form someone who does.
